packages/p2p-media-loader-core/src/http-loader.ts

```diff
@@ -50,10 +53,18 @@ export class HttpRequestExecutor {
   private async fetch() {
     const { segment } = this.request;
     try {
-      const response = await window.fetch(segment.url, {
+      const request = new Request(segment.url, {
         headers: this.requestHeaders,
         signal: this.abortController.signal,
       });
+
+      this.settings.httpRequestSetup?.(
+        segment.url,
+        request,
```

Comment:
`Request` object can't be configured as far as I know. As discussed, did you test the solution by adding an arbitrary header?

Reply:
What about `credentials` field? It can be definitely set via init constructor parameter but looks like can't be changed on `Request` object

Reply:
This code prints `same-origin`:
```js
const myRequest = new Request("flowers.jpg");
myRequest.credentials = 'omit';
console.log(myRequest.credentials);
```
But this code prints `omit`:
```js
const myRequest = new Request("flowers.jpg", {
  credentials: 'omit'
});
console.log(myRequest.credentials);
```